File: frontend/src/types/question.ts
export enum DifficultyLevel {
  EASY = 'easy',
  MEDIUM = 'medium',
  HARD = 'hard'
}

export interface TestCase {
  input: string;
  output: string;
  is_sample: boolean;
}

export interface Question {
  id: number;
  title: string;
  description: string;
  difficulty: DifficultyLevel;
  points: number;
  time_limit: number;
  memory_limit: number;
  constraints?: string;
  test_cases: TestCase[];
  examples?: any[];
  hints?: string[];
  starter_code?: Record<string, string>;
  teacher_id: number;
}

export interface QuestionCreate {
  title: string;
  description: string;
  difficulty: DifficultyLevel;
  points: number;
  time_limit: number;
  memory_limit: number;
  constraints?: string;
  test_cases: TestCase[];
  examples?: any[];
  hints?: string[];
  starter_code?: Record<string, string>;
}

File: backend/app/models/question.py
from sqlalchemy import Column, Integer, String, Text, ForeignKey, JSON, Enum
from sqlalchemy.orm import relationship
import enum
from app.database import Base

class DifficultyLevel(enum.Enum):
    EASY = "easy"
    MEDIUM = "medium"
    HARD = "hard"

class Question(Base):
    __tablename__ = "questions"
    
    id = Column(Integer, primary_key=True, index=True)
    title = Column(String, nullable=False)
    description = Column(Text, nullable=False)
    difficulty = Column(Enum(DifficultyLevel), default=DifficultyLevel.EASY)
    points = Column(Integer, default=10)
    time_limit = Column(Integer, default=2)  # seconds
    memory_limit = Column(Integer, default=256)  # MB
    
    # Test cases stored as JSON
    test_cases = Column(JSON)  # [{"input": "", "output": "", "is_sample": true}]
    
    # Constraints and examples
    constraints = Column(Text)
    examples = Column(JSON)
    
    # AI-assisted features
    hints = Column(JSON)
    starter_code = Column(JSON)  # {"python": "", "javascript": "", "java": ""}
    
    teacher_id = Column(Integer, ForeignKey("users.id"))
    
    # Relationships
    teacher = relationship("User", back_populates="questions")
    submissions = relationship("Submission", back_populates="question")